### DISCUSSION TITLE
Is bogo_shim with boringssl still in use?

### DISCUSSION THREAD

**Discussion by pstanisz** [randombit/botan#3797] [General]
Hi,

Recently I tried to launch bogo_shim with botan 3.0.0 and boringssl fork (rene/runner-20230705 branch), as I did it with 2.x.x release in the past.
But now I'm getting all tests marked as UNIMPLEMENTED, i.e.:
UNIMPLEMENTED (FragmentAlert-DTLS)

Are boringssl & bogo_shim still in use?
I noticed that TLS-Anvil is now used for testing of TLS functionality and thought that maybe boringssl was abandoned?

Best regards,
Piotr

**Comment by reneme** [comment]
Hi Piotr,

we do continue to test Botan's TLS implementation with boringssl's test suite and there are no plans to abandon them. The TLS Anvil tests were added additionally.

If you happen to use Visual Studio Code, the easiest way to run them is as follows:

1. Configure and build
```bash
./configure.py --build-targets=static,bogo_shim --without-documentation --build-tool=make
make -j8
```
2. Symlink the VS code config into the repo root and launch it:
```bash
ln -s src/editors/vscode .vscode
code .
```
3. Run the tests from within the IDE
```
Menu: Terminal > Run Task... > "Run BoGo Tests"
```

If you don't use VS Code, you can also run the python script behind this IDE task straight from the command line. It assumes that Botan is built as shown above and automatically takes care of setting up the appropriate boringssl fork.
```bash
python3 src/editors/vscode/scripts/bogo.py
```

**Reply by pstanisz** [reply]
Hi René,

Thank you for the hints. It works and is really nicely integrated with VS code now!
Very useful and convenient improvement for all VS code users.

BR,
Piotr

**Reply by reneme** [reply]
FWIW: With #3798 the build commands above would now also work with `--build-tool=ninja`.
